### PR TITLE
Updating the ARM HWIntrinsic test generation script to automatically split the tests

### LINKED ISSUE #37282: Break up AdvSimd test
This test is comprised of over 1200 .cs files. As a result, it regularly times out in GCStress runs, an issue tracked by https://github.com/dotnet/runtime/issues/33948.

Note that the CI has a 60 minute per-test timeout and 240 minute per-collection timeout for GCStress jobs (see eng\common\templates\runtimes\run-test-job.yml).

This test should be broken up to keep it to a manageable size that would run GCStress in less time.

### PR DESCRIPTION
**Pull request by tannergooding**
This updates the `GenerateTests.csx` file to automatically split ISAs into groups of no more than `x` tests. `x` is currently `256` as that is approx the number of tests in `x86.Avx2` which has never had issues and should be small enough to not hit any timeout issues.

This resolves https://github.com/dotnet/runtime/issues/37282

### CODE REVIEW

**Comment by tannergooding** [comment]
CC. @jkotas, @BruceForstall, @CarolEidt, @echesakovMSFT

**Review by echesakov — APPROVED** [review]
@tannergooding Thanks for fixing the issue! Originally, I though about splitting the tests by the method name - that way it will be easier to debug and reproduce the failures.

However, given the urgency for the fix - let's merge your PR first and I will follow up later.

**Comment by tannergooding** [comment]
Going to merge to unblock CI. I'll follow up with any remaining feedback that comes after merge.

**Pull request merged**
